src/storage-manager.coffee

```diff
@@ -67,6 +61,9 @@ class _StorageManager
     collection.on 'remove', (model) ->
       model.invalidateCache()
 
+    collection.on 'change', (model) =>
+      @_invalidateCache(model.brainstemKey) unless model.isNew()
```

Comment:
Yes. Adding/resetting models on the storage manager collection should not be indicate busting the cache. 

Removing as a result of a deletion API should bust the cache as well. I think this can be fixed by adding extra logic to the association layer: adding / removing an association ID should bust the cache of the association type.